File: japiwznm/WznmQAppList.java
/**
  * \file WznmQAppList.h
  * Java API code for record of table TblWznmQAppList
  * \author Alexander Wirthmueller
  * \date created: 27 Aug 2020
  * \date modified: 27 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQAppList {
	
	public WznmQAppList(
				int jnum
				, String stubGrp
				, String stubOwn
				, String Short
				, String Title
				, String srefIxWznmVApptarget
				, String titIxWznmVApptarget
				, String stubVerRefWznmMVersion
			) {
		this.jnum = jnum;
		this.stubGrp = stubGrp;
		this.stubOwn = stubOwn;
		this.Short = Short;
		this.Title = Title;
		this.srefIxWznmVApptarget = srefIxWznmVApptarget;
		this.titIxWznmVApptarget = titIxWznmVApptarget;
		this.stubVerRefWznmMVersion = stubVerRefWznmMVersion;
	};

	public int jnum;
	public String stubGrp;
	public String stubOwn;
	public String Short;
	public String Title;
	public String srefIxWznmVApptarget;
	public String titIxWznmVApptarget;
	public String stubVerRefWznmMVersion;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQAppList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			stubGrp = Xmlio.extractStringUclc(doc, basexpath, "stubGrp", "grp", null, 0);
			stubOwn = Xmlio.extractStringUclc(doc, basexpath, "stubOwn", "own", null, 0);
			Short = Xmlio.extractStringUclc(doc, basexpath, "Short", "sho", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			srefIxWznmVApptarget = Xmlio.extractStringUclc(doc, basexpath, "srefIxWznmVApptarget", "trg", null, 0);
			titIxWznmVApptarget = Xmlio.extractStringUclc(doc, basexpath, "titIxWznmVApptarget", "trg2", null, 0);
			stubVerRefWznmMVersion = Xmlio.extractStringUclc(doc, basexpath, "stubVerRefWznmMVersion", "ver", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQAppList.cpp
/**
	* \file WznmQAppList.cpp
	* API code for table TblWznmQAppList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 27 Aug 2020
	* \date modified: 27 Aug 2020
	*/

#include "WznmQAppList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQAppList
 ******************************************************************************/

WznmQAppList::WznmQAppList(
			const uint jnum
			, const string stubGrp
			, const string stubOwn
			, const string Short
			, const string Title
			, const string srefIxWznmVApptarget
			, const string titIxWznmVApptarget
			, const string stubVerRefWznmMVersion
		) {
	this->jnum = jnum;
	this->stubGrp = stubGrp;
	this->stubOwn = stubOwn;
	this->Short = Short;
	this->Title = Title;
	this->srefIxWznmVApptarget = srefIxWznmVApptarget;
	this->titIxWznmVApptarget = titIxWznmVApptarget;
	this->stubVerRefWznmMVersion = stubVerRefWznmMVersion;
};

bool WznmQAppList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQAppList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "stubGrp", "grp", stubGrp);
		extractStringUclc(docctx, basexpath, "stubOwn", "own", stubOwn);
		extractStringUclc(docctx, basexpath, "Short", "sho", Short);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "srefIxWznmVApptarget", "trg", srefIxWznmVApptarget);
		extractStringUclc(docctx, basexpath, "titIxWznmVApptarget", "trg2", titIxWznmVApptarget);
		extractStringUclc(docctx, basexpath, "stubVerRefWznmMVersion", "ver", stubVerRefWznmMVersion);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQAppList
 ******************************************************************************/

ListWznmQAppList::ListWznmQAppList() {
};

ListWznmQAppList::ListWznmQAppList(
			const ListWznmQAppList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQAppList(*(src.nodes[i]));
};

ListWznmQAppList& ListWznmQAppList::operator=(
			const ListWznmQAppList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQAppList(*(src.nodes[i]));

	return *this;
};

ListWznmQAppList::~ListWznmQAppList() {
	clear();
};

void ListWznmQAppList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQAppList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQAppList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQAppList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQAppList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQAppList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQAppList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

